### PR TITLE
Model monitoring and detecting drifts in machine learning models using Deepchecks

### LINKED ISSUE #7248: [Machine Learning] Model monitoring and detecting drifts in machine learning models using Deepchecks
#### Proposed title of article
[Machine Learning] Model monitoring and detecting drifts in machine learning models using Deepchecks

#### Proposed article introduction
Model monitoring is an operational stage in the machine learning lifecycle that comes after model deployment. It entails monitoring your ML models for changes such as model degradation, data drift, and concept drift, and ensuring that the model is maintaining an acceptable level of performance. It is a close tracking of the performance of ML models in production so that production and AI teams can identify potential issues before they impact the business

Validation results during development will seldom fully justify your model's performance in production. This is a key reason why you have to monitor your models after deployment to make sure they keep performing as well as they're supposed to. A robust MLOps infrastructure should be able to proactively monitor for service health, data relevance, model performance, and business impact.

In predictive analytics and machine learning, we have model/concept drift and data drift. Model/concept drift means that the statistical properties of the target variable, which the model is trying to predict, change over time in unforeseen ways. This causes problems because the predictions become less accurate as time passes. Model drift is caused due to the degradation of model performance due to changes in data and relationships between input and output variables. It is relatively common for model drift to impact an organization negatively over time or sometimes suddenly.

Data-drift is defined as a variation in the production data from the data that was used to test and validate the model before deploying it in production. 

Deepchecks is a Python package for comprehensively validating your machine learning models and data with minimal effort. This includes checks related to various types of issues, such as model performance, data integrity, and distribution mismatches. It checks the ML models for things like errors, crashes, and latency, but most importantly, to ensure that your model is maintaining a predetermined desired level of performance.

#### Key takeaways 
1. What Is Model Monitoring?
2. Why Is Model Monitoring Important?
3. What is model/concept drift?
4. What is data drift?
5. What is Deepchecks?
6. Using Deepchecks for Offline ML Data Drift Detection.
7. Running a Deepchecks Full Suite (Model Error Analysis, Label Ambiguity, and Data Sample Leakage.)
8. Dataset Integrity Checks using Deepchecks.

#### Article quality
In this tutorial, we will cover the concept of model monitoring and why it is important to monitor the model in production. We will also discuss what you should track in the model during production to detect changes that may affect the model performance. We will also implement a machine learning model using Python. We will then monitor the model using Deepchecks to detect the concept and data drifts by running a Deepchecks full Suite. We will explain all the Deepchecks concepts and implement them practically in Google Colab.

The tutorial will be detailed and it will have everything for the reader to get started with model monitoring and handling drifts in the model.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by simonndiritu477**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #7248

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

**Comment by Bayler** [comment]
@simonndiritu477 There are some grammatical mistakes that the basic version of Grammarly fail to detect. Please pass your article through Grammarly premium and correct those mistakes.

**Comment by simonndiritu477** [comment]
Hello @Bayler, I have performed a Grammarly check. Thanks.

File: content/articles/model-monitoring-and-detecting-drifts-in-machine-learning-models-using-deepchecks/index.md [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+- [Loading bank customer dataset](#loading-bank-customer-dataset)
+- [Selecting input and output variables from the columns](#selecting-input-and-output-variables-from-the-columns)
+- [Splitting the bank customer dataset](#splitting-the-bank-customer-dataset)
+- [Importing the Pipeline module](#importing-the-pipeline-module)
+- [Fitting the Pipeline](#fitting-the-pipeline)
+- [Accuracy score for the customer classification model](#accuracy-score-for-the-customer-classification-model)
+- [Implementing Deepchecks](#implementing-deepchecks)
+- [Creating the two dataset objects](#creating-the-two-dataset-objects)
+- [Importing the `full_suite` method](#importing-the-fullsuite-method)
+- [The `full_suite` outputs](#the-fullsuite-outputs)
```
Comment (Bayler) [inline content/articles/model-monitoring-and-detecting-drifts-in-machine-learning-models-using-deepchecks/index.md:38]: Please fix this headings. They aren't rendering.

**Comment by Bayler** [comment]
Consider shortening this URL `model-monitoring-and-detecting-drifts-in-machine-learning-models-using-deepchecks`. It is too long.

File: content/articles/model-monitoring-and-detecting-drifts-in-machine-learning-models-using-deepchecks/index.md [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+![Deep checks metods](/engineering-education/model-monitoring-and-detecting-drifts-in-machine-learning-models-using-deepchecks/deepchecks-methods.png)
+
+Even though we have listed all the methods, we will not implement all of them. We will use `Dataset`, `Suite`, and `Check`.
+
+#### Dataset method
+It creates two dataset objects to represent the original dataset. 
+
+The first one will represent the training dataset. The second one will represent the testing dataset. 
+
+It will transform the original dataset into a format that Deepchecks will understand. We will then add the transformed dataset (the two dataset objects) to Deepchecks for model drift detection.
```
Comment (Bayler) [inline content/articles/model-monitoring-and-detecting-drifts-in-machine-learning-models-using-deepchecks/index.md:201]: As a rule of thumb, make sure your paragraphs are at least three lines long

File: content/articles/model-monitoring-and-detecting-drifts-in-machine-learning-models-using-deepchecks/index.md [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+
+#### Suite method
+We will use this method this gives the general/model performance and summary. We will import `full_suite` from `Suite`. `full_suite` will perform an exhaustive check- it will perform multiple checks and detect any data drifts and concepts drifts. It also shows the confidence score of the model and the data set integrity.
```
Comment (Bayler) [inline content/articles/model-monitoring-and-detecting-drifts-in-machine-learning-models-using-deepchecks/index.md:204]: This sentence is not clear.

File: content/articles/model-monitoring-and-detecting-drifts-in-machine-learning-models-using-deepchecks/index.md [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+It uses the `run` function to run the method. It will then analyze the Deepchecks dataset objects to detect concept drift.
+
+To see the output, input this code:
+
+```python
+concept_drift_output
+```
+It gives the following output:
+
+![Checkinge concept drift](/engineering-education/model-monitoring-and-detecting-drifts-in-machine-learning-models-using-deepchecks/checking-concept-drift.png)
```
Comment (Bayler) [inline content/articles/model-monitoring-and-detecting-drifts-in-machine-learning-models-using-deepchecks/index.md:390]:
```suggestion
![Checking concept drift](/engineering-education/model-monitoring-and-detecting-drifts-in-machine-learning-models-using-deepchecks/checking-concept-drift.png)
```

**Comment by Bayler** [comment]
@simonndiritu477 Please work on those revisions before we can proceed for final review.

**Comment by simonndiritu477** [comment]
Hello @Bayler, I have worked on the revisions.

**Comment by hectorkambow** [comment]
jumping on this one

**Pull request merged**